tests/test_client_functional.py

```diff
@@ -2453,6 +2458,53 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url, error_message_url, expected_exception_class",
+    (
+        # yarl.URL.__new__ raises ValueError
+        ("http://:/", "http://:/", InvalidUrlRedirectClientError),
+        (
+            "http://example.org:non_int_port/",
+            "http://example.org:non_int_port/",
+            InvalidUrlRedirectClientError,
+        ),
+        # # yarl.URL.origin raises ValueError
+        ("http:/", "http:///", InvalidUrlRedirectClientError),
+        ("http:/example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("http:///example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("bluesky://d:i:d", "bluesky://d:i:d", InvalidUrlRedirectClientError),
+        ("call:+3801234567", r"call:\+3801234567", NonHttpUrlRedirectClientError),
+        ("skype:handle", "skype:handle", NonHttpUrlRedirectClientError),
+        (
+            "slack://instance/room",
+            "slack://instance/room",
+            NonHttpUrlRedirectClientError,
+        ),
+        ("steam:code", "steam:code", NonHttpUrlRedirectClientError),
+        ("twitter://handle", "twitter://handle", NonHttpUrlRedirectClientError),
+    ),
+)
+async def test_invalid_redirect_url(
+    aiohttp_client: Any,
+    invalid_redirect_url: Any,
+    error_message_url: str,
+    expected_exception_class: Any,
+) -> None:
+    headers = {hdrs.LOCATION: invalid_redirect_url}
+
+    async def handler_request(request):
+        return web.Response(status=301, headers=headers)
+
+    app = web.Application()
+    app.router.add_get("/redirect", handler_request)
```

Comment:
Let's follow the structure of other test here:
```suggestion
    async def generate_redirecting_response(request):
        raise web.HTTPFound(location=invalid_redirect_url)

    app = web.Application()
    app.router.add_get("/redirect", generate_redirecting_response)
```

Reply:
@webknjaz I applied this but it's not response - it's error and some of test cases don't raise expected exceptions. I Restored it for now